### ISSUE TITLE
Fix `named-grid-areas-no-invalid` false positive for interpolation in styled components

### ISSUE THREAD

**Issue by RawToaster** [stylelint/stylelint#8597]
### What minimal example or steps are needed to reproduce the bug?

Create a styled component with `grid-template-areas` property that has a conditional value where a string literal is in the condition.

```ts
export const Foo = styled.div<{ $labelAlignment: 'middle' | 'right' }>`
  display: grid;
  
  grid-template-areas: ${({ $labelAlignment }) => ($labelAlignment === 'middle' ? "'back label .'" : "'back . label'")};

  grid-template-columns: auto auto auto;
`
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "named-grid-areas-no-invalid": true
  }
}
```

### How did you run Stylelint?

```sh
stylelint --max-warnings=0 .
```

### Which Stylelint-related dependencies are you using?

```json
{
  "postcss-styled-syntax": "^0.7.1",
  "stylelint": "^16.20.0",
  "stylelint-config-standard": "^38.0.0",
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

```sh
src/style.ts
  9:83  ✖  Expected same number of cell tokens in each string  named-grid-areas-no-invalid

✖ 1 problem (1 error, 0 warnings)
```

### Do you have a proposal to fix the bug?

Seems like the issue is the string literal in the condition is being checked against the possible return values.

**Comment by ybiquitous** [comment]
@RawToaster Thanks for the report. Could you reproduce it on our online demo <https://stylelint.io/demo>?

**Comment by RawToaster** [comment]
@ybiquitous Yes I can! Here is the [link](https://stylelint.io/demo/#N4Igxg9gJgpiBcICWBbADhATgFwAQGdsBPAGxilwDNMIVcByQ08gWknQgDsZPt96AOpyGROhXAENcAXgLEyUAHRQkANwA8wACSUIEeHMxJOAcwC+APgAGQk0agtsMdCQlOWEzDAn4DW4AAUwLg6erhmAJQyFiG6EDLSsvQSnESGxia4AO5I2AAWuCqUlDBevLicAK4oAEaluBCUuNgQANY8+M15Kbhe2JWYnLiqEiSVMPy4APy4AiDJkpKCILgGc-Q1uJs1y2YA3EI2wpwA9Ce4AKIAHhIuMA1NuRUQePh5EDmmXfelNJjZeR4khIJHSpk63VU93wt3uVVq9UazTaHREXHEm1kTAUyjUmlC+jB5mstnsjmcaFc7k83l8IUCwQJ4Si0hiTMSSTAuC5YHo01m8ykQuWqwFGy2W12B04VhAABoQJQkGQAHKwhAgGA3O6KPjy8BcJUmDXAIS4AVapycKD4OYGADaZvNAuxMBIxmwbENSBMLEIKSgnigcydAF05U65mBKoRaABlIi8CRXO0CjCEMD4fB++SsfCJ7DJkOcMz60RGgBiWBQbg1ACt8Fx9bA0PgTZGQK73bxU3MqRNsHMI0MXbnu57yz6cwGg72QP3CEOO66HOwMNxeLaEAKAHoANkUAEYjwB2Jcjubp7CZ7Mrv0FovbuY7gAMihPR+LpbMQA), and if that doesn't work, I'll paste it below as well.


example.ts:
```ts
import styled from 'styled-components'

const a = styled.div<{$foo: string}>`
grid-template-areas: ${({ $foo }) => $foo === 'any string with different number of tokens than return values' ? "'a a a'" : "'b b b'"};
`

// Example of it not showing the error when all strings have same number of tokens
const b = styled.div<{$foo: string}>`
grid-template-areas: ${({ $foo }) => $foo === 'c c c' ? "'a a a'" : "'b b b'"};
`
```

config:
```json
{
  "extends": [
    "stylelint-config-standard"
  ],
  "customSyntax": "postcss-styled-syntax"
}
```

dependencies:
```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest",
  "styled-components": "^6.1.17",
  "postcss-styled-syntax": "^0.7.1"
}
```

**Comment by shahana308** [comment]
@ybiquitous I'd love to work on this! Let me know if anyone's already started, or I'll go ahead and open a PR
